features/steps/steps.py

```diff
@@ -283,4 +294,19 @@ def step_impl(context, representation_id):
                     errors.append(representation_shape_error(inst, representation_id))
     
     handle_errors(context, errors)
+
+@then('Each {entity} may be nested by only the following entities: {other_entities}')
+def step_impl(context, entity, other_entities):
+
+    allowed_entity_types = other_entities.split(', ')
+
+    errors = []
+    if getattr(context, 'applicable', True):
+        for inst in context.model.by_type(entity):
+            nested_entities = [i for rel in inst.IsNestedBy for i in rel.RelatedObjects]
+            nested_entity_types = [i.is_a() for i in nested_entities]
+            if not set(nested_entity_types) <= set((allowed_entity_types)):
+                differences = list(set(nested_entity_types) - set(allowed_entity_types))
```

Comment:
No need to convert to list. With the comments above taken into account (sets already) then `nested_entity_types - allowed_entity_types` suffices.

Reply:
https://github.com/buildingSMART/ifc-gherkin-rules/commit/433fe4ca8d9318574873a26542a6f321856aacd1